### PR TITLE
Add alternative Statement syntax to suppress newlines

### PR DESCRIPTION
**Pull request by craigpepper**
Add alternative Statement closing delimiter syntax `-%}` that suppresses in the output an immediately following newline.
This PR addresses Issue #49.

### CODE REVIEW

File: src/inja.hpp [hunk 1]
```diff
@@ -645,7 +645,7 @@ class Parser {
 	}
 
 	std::map<Parsed::Delimiter, Regex> regex_map_delimiters = {
-		{Parsed::Delimiter::Statement, Regex{"\\{\\%\\s*(.+?)\\s*\\%\\}"}},
+		{Parsed::Delimiter::Statement, Regex{"\\{\\%\\s*(.+?)\\s*(\\%\\}|\\-\\%\\}\\s*\\n)"}},
```
Comment (flexferrum) [inline src/inja.hpp]: Does it handle cases like this:
`
Hello {% if var -%}  {{ var1 }} {% endif %}
`
The space between `-%}` and next `{{` should be removed.
http://jinja.pocoo.org/docs/2.10/templates/#whitespace-control

Comment (craigpepper) [reply src/inja.hpp]: It didn't, but I just pushed an update that does.

**Comment by flexferrum** [comment]
I have thoughts not to do it via regexp, but 'manually' check the symbol right before '%}' or '}}' and skip whitespaces if necessary.

File: src/inja.hpp (hunk 1 above)
Comment (craigpepper) [reply src/inja.hpp]: So what seemed a simple change of making the newline optional in the regex had the unintended side effect of also removing whitespace at the beginning of the next line, after the newline.  I clearly have an imperfect understanding of ECMAScript regexes.
I have reverted back to the original commit that just removes whitespace up to and including the newline.  The intent of this PR is to address Issue #49, and the original commit does that.

**Comment by craigpepper** [comment]
I agree, it seems like it would take something beyond regex to achieve platform independent jinja-like whitespace handling via `{%- ... -%}` etc - at least until C++17 regex multiline support becomes common.  However, inja is at its core a regex-based parser, so it seems quite difficult to retrofit manual checking.

File: src/inja.hpp (hunk 1 above)
Comment (pantor) [reply src/inja.hpp]: Jinja also removes the whitespace in the next line. So I would just delete the `\\n` in the regex, and then we are good to go.

**Pull request closed without merging**